app/controllers/uploads_controller.rb

```diff
@@ -0,0 +1,43 @@
+require 'zip'
+
+class UploadsController < ApplicationController
+
+  def index
+    @libraries = Library.all
+  end
+
+  def upload_zip
```

Comment:
In line with the comment on `routes.rb`, this could just be the default `create` action.